tests/datasource/data_connector/test_data_connector_util.py

```diff
@@ -231,6 +231,42 @@ def test_convert_data_reference_string_to_batch_identifiers_using_regex():
     )
 
 
+def test_convert_data_reference_string_to_batch_identifiers_using_regex_with_named_groups(
+    caplog,
+):
+    data_reference = "alex_20200809_1000.csv"
+    pattern = r"^(?P<name>.+)_(?P<timestamp>\d+)_(?P<price>\d+)\.csv$"
+    group_names = ["name", "timestamp", "price"]
+    assert convert_data_reference_string_to_batch_identifiers_using_regex(
+        data_reference=data_reference, regex_pattern=pattern, group_names=group_names
+    ) == (
+        "DEFAULT_ASSET_NAME",
+        IDDict(
+            {
+                "name": "alex",
+                "timestamp": "20200809",
+                "price": "1000",
+            }
+        ),
+    )
+
+    data_reference = "alex_20200809_1000.csv"
```

Comment:
How do you feel about keeping the same `data_reference` and `pattern` vars defined at the top of this test func and just use "cost" in the in the updated `group_names` list?

If you do that, you would also need to change `cost` to `price` in the asserted error message

Reply:
I like that!